### DISCUSSION TITLE
Support for Separate boot filesystem on root?

### DISCUSSION THREAD

**Discussion by sm-tradeboox** [zbm-dev/zfsbootmenu#313] [General]
Hello. 

As I understand it, for ZBM to work properly, each root DS has to have a /boot directory containing the kernel(s) and initramfs images. Can I have /boot as a separate ZFS filesystem - i.e. ```zroot/ROOT/os/boot```? Can /boot be an encrypted FS? 

I'm just trying to see what my options are.

Thanks.

**Comment by ahesford** [comment]
No, ZFSBootMenu defines a "boot environment" as a filesystem that mounts at `/` and contains a `/boot` directory with kernels. See https://github.com/zbm-dev/zfsbootmenu/blob/master/docs/BOOTENVS.md for more details.
